rts/Lua/LuaOpenGL.cpp

```diff
@@ -4657,15 +4660,288 @@ int LuaOpenGL::GetMapRendering(lua_State* L)
+			return 1;
+		} break;
+		case hashString("splatDetailTexture"): {
+			CSMFReadMap* smfMap = (CSMFReadMap*)readMap;
+			lua_pushboolean(L, smfMap->GetSplatDistrTexture() != 0 &&
+			                   smfMap->GetSplatDetailTexture() != 0);
+			return 1;
+		} break;
+		case hashString("splatDetailNormalTexture"): {
+			CSMFReadMap* smfMap = (CSMFReadMap*)readMap;
+			lua_pushboolean(L, smfMap->GetSplatDistrTexture() != 0 &&
+			                   smfMap->HaveSplatNormalTexture());
+			return 1;
+		} break;
 		case hashString("splatDetailNormalDiffuseAlpha"): {
 			lua_pushboolean(L, mapRendering->splatDetailNormalDiffuseAlpha);
 			return 1;
 		} break;
+		case hashString("waterAbsortion"): {
+			CSMFReadMap* smfMap = (CSMFReadMap*)readMap;
+			lua_pushboolean(L, smfMap->HasVisibleWater());
+			return 1;
+		} break;
+		case hashString("skyReflection"): {
+			CSMFReadMap* smfMap = (CSMFReadMap*)readMap;
+			lua_pushboolean(L, smfMap->GetSkyReflectModTexture() != 0);
+			return 1;
+		} break;
+		case hashString("blendNormals"): {
+			CSMFReadMap* smfMap = (CSMFReadMap*)readMap;
+			lua_pushboolean(L, smfMap->GetBlendNormalsTexture() != 0);
+			return 1;
+		} break;
+		case hashString("lightEmission"): {
+			CSMFReadMap* smfMap = (CSMFReadMap*)readMap;
+			lua_pushboolean(L, smfMap->GetLightEmissionTexture() != 0);
+			return 1;
+		} break;
+		case hashString("parallaxMapping"): {
+			CSMFReadMap* smfMap = (CSMFReadMap*)readMap;
+			lua_pushboolean(L, smfMap->GetParallaxHeightTexture() != 0);
+			return 1;
+		} break;
+		// float
+		case hashString("baseDynamicMapLight"): {
+			CSMFReadMap* smfMap = (CSMFReadMap*)readMap;
+			CSMFGroundDrawer* groundDrawer = (CSMFGroundDrawer*)smfMap->GetGroundDrawer();
+			const GL::LightHandler* lightHandler = groundDrawer->GetLightHandler();
+			lua_pushnumber(L, lightHandler->GetBaseLight());
+			return 1;
+		} break;
+		case hashString("maxDynamicMapLight"): {
+			CSMFReadMap* smfMap = (CSMFReadMap*)readMap;
+			CSMFGroundDrawer* groundDrawer = (CSMFGroundDrawer*)smfMap->GetGroundDrawer();
+			const GL::LightHandler* lightHandler = groundDrawer->GetLightHandler();
+			lua_pushnumber(L, lightHandler->GetMaxLights());
+			return 1;
+		} break;
 	}
 
 	luaL_error(L, "[%s] unknown key %s", __func__, key);
 	return 0;
 }
 
+int LuaOpenGL::GetMapShaderUniform(lua_State* L)
+{
+	const int args = lua_gettop(L); // number of arguments
+	if ((args != 1) || !lua_istable(L, 1)) {
+		luaL_error(L, "Incorrect arguments to gl.GetMapShaderUniform(table)");
+	}
+
+	// Store the arguments in a list
+	std::vector<string> keys;
+	const int table = lua_gettop(L);
+	for (lua_pushnil(L); lua_next(L, table) != 0; lua_pop(L, 1)) {
+		if (!lua_israwstring(L, -1)) {
+			LOG_L(L_WARNING, "gl.GetMapShaderUniform: bad uniform name type");
+			return 0;
+		}
+		keys.push_back(lua_tostring(L, -1));
+	}
+
+	lua_createtable(L, keys.size(), 0);
+	int count = 1;
+	for (auto key : keys) {
+		if (key == "diffuseTex") {
```

Comment:
Use `switch (hashString(key)) { ... }` here, as in `GetMapRendering`.

Reply:
Done